src/foam/u2/view/ReadOnlyEnumView.js

```diff
@@ -9,15 +9,17 @@ foam.CLASS({
   name: 'ReadOnlyEnumView',
   extends: 'foam.u2.View',
 
+  requires: ['foam.u2.view.EnumBadgeView'],
+
   documentation: 'A read-only view for foam.core.Enum properties.',
 
   methods: [
     function initE() {
       this.SUPER();
 
-      this.add(this.slot((data) => {
+      this.add(this.slot(data => {
         if ( ! data ) return '';
-        return this.E().addClasses(data.classes()).style(data.toStyle()).add(data.label);
+        return this.E().add(foam.u2.view.EnumBadgeView.create({ data: data }));
```

Comment:
Could you just:

return foam.u2.view.EnumBadgeView.create({ data: data })

?

Reply:
Aren't the EnumBadView and ReadOnlyEnumViews really the same thing now?

Reply:
Technically they are, but I don't want to get rid of ReadOnlyEnumView in case it is being used somewhere else.

Reply:
We could replace it with EnumBadgeView and it should work fine.